compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2965,6 +2965,42 @@ declare_lint! {
     };
 }
 
+declare_lint! {
+    /// The `ptr_to_int_cast` lint checks casting pointers to integers other than `usize`. These
+    /// casts are often bugs.
```

Comment:
```suggestion
    /// The `ptr_to_int_cast` lint triggers if a pointer is cast to any integer type other than
    /// `usize`, since doing so is often a bug.
```